lib/game.ex

```diff
@@ -3,37 +3,54 @@ defmodule Game do
             player_2: %Computer{}, 
             current_player: %Player{}, 
             winner: nil, 
-            board: %Board{}.spaces
+            board: %Board{}.spaces,
+            over: false
             
-  # convert to private functions?
-  def check_three([x, x, x]), do: true
-  def check_three([a, b, c]), do: false
+  defp check_three([x, x, x]), do: x
+  defp check_three([_a, _b, _c]), do: nil
 
-  # convert to private functions?
-  def check_line([head | tail]), do: [check_three(head) | check_line(tail)]
-  def check_line([]), do: []
+  defp check_line([head | tail]), do: [check_three(head) | check_line(tail)]
+  defp check_line([]), do: []
 
-  # convert to private functions?
-  def convert_to_columns([a, b, c, d, e, f, g, h, i]), do: [[a, d, g], [b, e, h], [c, f, i]]
-  def convert_to_diagonals([a, _b, c, _d, e, _f, g, _h, i]), do: [[a, e, i], [g, e, c]]
-  def convert_to_rows(board), do: Enum.chunk(board, 3)
+  defp convert_to_columns([a, b, c, d, e, f, g, h, i]), do: [[a, d, g], [b, e, h], [c, f, i]]
+  defp convert_to_diagonals([a, _b, c, _d, e, _f, g, _h, i]), do: [[a, e, i], [g, e, c]]
+  defp convert_to_rows(board), do: Enum.chunk(board, 3)
 
-  def game_over?(board) do
+  def get_winner(game) do
+    winner =
```

Comment:
Instead of a single statement with three or conditions, I'd like to see these assigned to clear variable names and then a separate statement to put them together.  For example:

```elixir
diagonal_winner = game.board |> convert_to_diagonals |> check_line |> Enum.find(&(&1))
row_winner = game.board |> convert_to_rows |> check_line |> Enum.find(&(&1))
column_winner = game.board |> convert_to_columns |> check_line |> Enum.find(&(&1))

winner = diagonal_winner || row_winner || column_winner
```

The benefit here is that the next person who dives into this will have sort of a living document in variable names to infer my intentions.  Otherwise, they'd just need to figure it out based on the code itself (which might have bugs).

Reply:
+ 1